lib/modules/datasource/aws-versioned-arn/index.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  LambdaClient,
+  LayerVersionsListItem,
+  ListLayerVersionsCommand,
+} from '@aws-sdk/client-lambda';
+import { cache } from '../../../util/cache/package/decorator';
+import { Lazy } from '../../../util/lazy';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, ReleaseResult } from '../types';
+
+export class AwsVersionedArnDataSource extends Datasource {
+  static readonly id = 'aws-versioned-arn';
+
+  override readonly caching = true;
+
+  private readonly lambda: Lazy<LambdaClient>;
+
+  constructor() {
+    super(AwsVersionedArnDataSource.id);
+    this.lambda = new Lazy(() => new LambdaClient({}));
+  }
+
+  @cache({
+    namespace: `datasource-${AwsVersionedArnDataSource.id}`,
+    key: (serializedArnFilter: string) =>
+      `getSortedLambdaLayerVersions:${serializedArnFilter}`,
+  })
+  async getSortedLambdaLayerVersions(
+    serializedLambdaLayerFilter: string
+  ): Promise<LayerVersionsListItem[]> {
+    const cmd = new ListLayerVersionsCommand({
+      LayerName: JSON.parse(serializedLambdaLayerFilter),
+    });
+
+    const matchingLayerVersions = await this.lambda.getValue().send(cmd);
+    matchingLayerVersions.LayerVersions =
+      matchingLayerVersions.LayerVersions ?? [];
+
+    return matchingLayerVersions.LayerVersions.sort((layer1, layer2) => {
+      return (layer1.Version ?? 0) - (layer2.Version ?? 0);
+    });
```

Comment:
I don't think we need to sort here, the upper layer will sort by version anyways.

Reply:
The AWS API does not say anything about the sort order. So I think it is better to sort. Or is sorting in general not needed?